app/models/adapters/payment_to_csv_adapter.rb

```diff
@@ -128,7 +128,10 @@ def term_dates(payment)
     end
 
 
-    def hips_info(payment)
+    def payment_processor_info(payment)
+      if payment.klarna_id.present?
+        return [payment.status, payment.klarna_id]
+      end
       [payment.status, payment.hips_id]
```

Comment:
I haven't tested it out locally but I'm not sure on why we have `payment.hips_id` here yet we return `payment.klarna_id` as part of the array on L#133.

Reply:
@BasilMawejje Good question ... we switched processors from HIPS to Klarna, but, we will have prior payments, in the DB, with a `hips_id` (and nil `klarna_id`).  So, we need to to return the correct ID based on the processor that was used for the payment.

Reply:
That explains it perfectly. Thanks 👍

Reply:
That explains it perfectly. Thanks 👍